numpy/ma/core.pyi

```diff
@@ -466,7 +468,39 @@ class MaskedArray(ndarray[_ShapeType_co, _DType_co]):
     def argmin(self, axis=..., fill_value=..., out=..., *, keepdims=...): ...
     def argmax(self, axis=..., fill_value=..., out=..., *, keepdims=...): ...
     def sort(self, axis=..., kind=..., order=..., endwith=..., fill_value=..., *, stable=...): ...
-    def min(self, axis=..., out=..., fill_value=..., keepdims=...): ...
+    @overload
+    def min(
+        self: _MaskedArray[_SCT],
+        axis: None = None,
+        out: None = None,
+        fill_value: _ScalarLike_co | None = None,
+        keepdims: Literal[False] | _NoValueType = ...,
+    ) -> _SCT: ...
+    @overload
+    def min(
+        self,
+        axis: _ShapeLike | None = None,
+        out: None = None,
+        fill_value: _ScalarLike_co | None = None,
+        keepdims: bool | _NoValueType = ...
+    ) -> Any: ...
+    @overload
+    def min(
+        self,
+        axis: None,
+        out: _MaskedArrayType,
+        fill_value: _ScalarLike_co | None = None,
+        keepdims: bool | _NoValueType = ...,
+    ) -> _MaskedArrayType: ...
+    @overload
+    def min(
+        self,
+        axis: _ShapeLike | None = None,
+        *,
+        out: _MaskedArrayType,
+        fill_value: _ScalarLike_co | None = None,
+        keepdims: bool | _NoValueType = ...,
+    ) -> _MaskedArrayType: ...
```

Comment:
At runtime this method violates the [LSP](https://wikipedia.org/wiki/Liskov_substitution_principle). And maybe you noticed already, but mypy and pyright don't like that, and will complain about it here. 

So we're violating the LSP intentionally and inevitably here. This justifies ignoring the errors that mypy and pyright report about it. See https://github.com/numpy/numtype/blob/a0e9fdf0c37292f5d5dae3a922cb0247925c5d9a/src/numpy-stubs/ma/core.pyi#L646 for how that can be done.

Reply:
thanks @jorenham - could you clarify what you mean that mypy and pyright don't like it please? what should I run to see them complaining?

Reply:
it's not part of the numpy toolchain, but it's what you'd see if you're using pylance and a mypy plugin for e.g. vscode.

Reply:
i don't see anything reported in vscode (where I have pylance), any particular configuration I need?

I've tried using `override` instead of `overload`, but then the tests fail:
```
E   AssertionError: Reveal mismatch at line 35
    
    error:
    Expression is of type "Any", not "numpy.bool[builtins.bool]"  [assert-type]
        assert_type(MAR_b.min(), np.bool)
```

Reply:
I'm guessing that you're using the default "basic" mode (horrible choice for a default imo). It should show up once you set it to "standard" or "strict"

Reply:
Also, `@override` isn't a replacement for `@overload`. It can be used to mark methods (or each overload of a method) as overriding a base class method. But at the moment, mypy will crash if do do that, so it's best not to use it (in combination with `@overload`).

Reply:
looks like i've got strict:

![image](https://github.com/user-attachments/assets/d6fb365f-0631-4792-92c6-27dc62bb531f)

either way, changing `overload` to `override` from `typing-extensions` fails the tests, any suggestions on how to handle that?

Reply:
Ah I see... Pyright's `reportIncompatibleMethodOverride` rule has already bene disabled at the top of this file.

But mypy isn't disabled, and it will indeed complain:
![image](https://github.com/user-attachments/assets/5ab178cd-a906-4995-984b-a63e253fab73)

Reply:
got it, thanks!